src/main/java/com/puppycrawl/tools/checkstyle/XMLLogger.java

```diff
@@ -129,9 +129,16 @@ public void addError(AuditEvent event) {
             writer.print(" message=\""
                 + encode(event.getMessage())
                 + "\"");
-            writer.println(" source=\""
-                + encode(event.getSourceName())
-                + "\"/>");
+            if (event.getModuleId() == null) {
+                writer.println(" source=\""
+                        + encode(event.getSourceName())
+                        + "\"/>");
+            }
+            else {
+                writer.println(" id=\""
```

Comment:
Why is this surrounded by an `if` if the attributes for both are different. We can have both in XML file if we wish.

Since we are completely overriding the `source` in the CLI output to only show the ID, we should do the exact same in the XML.
Make this attribute name `source` and not `id`.